x-pack/legacy/plugins/uptime/index.ts

```diff
@@ -33,6 +34,13 @@ export const uptime = (kibana: any) =>
       home: ['plugins/uptime/register_feature'],
     },
     init(server: KibanaServer) {
-      initServerWithKibana(server);
+      const initializerContext = {} as PluginInitializerContext;
+      const { elasticsearch, xpack_main } = server.plugins;
+      plugin(initializerContext).setup({
+        elasticsearch,
+        usageCollector: server.usage,
+        xpack: xpack_main,
+        route: (arg: any) => server.route(arg),
```

Comment:
Not critical as a first step, but at some point it would be good to switch to using the real NP core router inside your shim. You can use `server.newPlatform.setup.core.http.createRouter()`.